### DISCUSSION TITLE
Is it possible to specify the Locale in DataFetchingEnvironment using the Accept-Language header?

### DISCUSSION THREAD

**Discussion by aerospike2010** [Netflix/dgs-framework#1649] [General]
Is it possible to specify the Locale in DataFetchingEnvironment using the Accept-Language header? 

I need this for using the graphql-dgs-extended-validation library, where I have to specify the Locale for the error message.

**Comment by congotej** [comment]
Hi @aerospike2010! If I understand the question correctly, you want to use a header value in your data fetcher. You can use [@requestHeader](https://netflix.github.io/dgs/datafetching/#requestheader-requestparam-and-cookievalue). This will pass along the value of said header to the data fetcher and you can use it however (i.e the Accept-Language header).

**Reply by djlfb** [reply]
Reviving this thread as currently there is potential for error by developers using `DataFetchingEnvironment#getLocale` rather than through the request header annotation or custom context.

If the `ExecutionInput` could be customised to allow the `Locale` in `DataFetchingEnvironment` to be explicitly set, that would avoid such innocent mistakes.

We could introduce linting, but I don't see any harm in making this customisable as well.

**Reply by srinivasankavitha** [reply]
Thanks for the discussion so fa. I would recommend opening a discussion in graphql-java directly for this feature since it is something that would be beneficial to support in DataFetchingEnvironment rather than customizing it for just the framework.
